### DISCUSSION TITLE
Where can I find the source code of Visual Studio Code's extension Dev Containers?

### DISCUSSION THREAD

**Discussion by tnmoon** [devcontainers/community#69] [Q&A] [answered]
Dev Containers, AKA remote-containers, is an extension of Visual Studio Code. But there is not any source code in it's [GitHub repository](https://github.com/Microsoft/vscode-remote-release). So, where can I get it's source code?

I have searched all around Google, but there are not any clues found. The only things I can find related to source code are the remote-containers extension source files in ~/.vscode path, but it is obfuscated.

**Comment by schlich** [comment]
I recall seeing a while back that it is not open source

**Comment by Destroy666x** [comment]
https://github.com/microsoft/vscode-dev-containers?

**Comment by bamurtaugh** [comment, marked as answer]
Thanks for opening this. The VS Code Dev Containers extension is not open source, and there's further info here: https://code.visualstudio.com/docs/remote/faq#_why-arent-the-remote-development-extensions-or-their-components-open-source.

The core CLI backing it is open source and available at https://github.com/devcontainers/cli.
